client/src/lib.rs

```diff
@@ -0,0 +1,55 @@
+use fuel_debugger::{Command, Response};
+use std::io;
+use std::net::SocketAddr;
+use tokio::io::{AsyncBufReadExt, AsyncWriteExt, BufReader};
+use tokio::net::{
+    tcp::{OwnedReadHalf, OwnedWriteHalf},
+    TcpListener,
+};
+
+pub mod names;
+
+pub struct Listener {
+    listener: TcpListener,
+}
+
+impl Listener {
+    pub fn new(listener: TcpListener) -> Self {
+        Self { listener }
+    }
+
+    pub async fn accept(&self) -> io::Result<(Client, SocketAddr)> {
```

Comment:
Why `SocketAddr` isn't an attribute of the `Client`? That way, we can avoid returning tuples